webapp/channels/src/components/channel_header/channel_header_title_favorite.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useRef, useCallback} from 'react';
+import {useSelector, useDispatch} from 'react-redux';
+import {FormattedMessage, useIntl} from 'react-intl';
+import OverlayTrigger from 'components/overlay_trigger';
+import Tooltip from 'components/tooltip';
+import {Constants } from 'utils/constants';
+import {favoriteChannel, unfavoriteChannel} from 'mattermost-redux/actions/channels';
+import {getCurrentChannel, isCurrentChannelFavorite} from 'mattermost-redux/selectors/entities/channels';
+
+type Props = {}
+
+const ChannelHeaderTitleFavorite = ({}: Props) => {
+    const intl = useIntl();
+    const dispatch = useDispatch()
+    const isFavorite = useSelector(isCurrentChannelFavorite);
+    const channel = useSelector(getCurrentChannel) || {};
+    const channelIsArchived = channel.delete_at !== 0;
+
+    const toggleFavoriteCallback = useCallback((e: React.MouseEvent<HTMLButtonElement>) => {
+        e.stopPropagation();
+        if (isFavorite) {
+            dispatch(unfavoriteChannel(channel.id));
+        } else {
+            dispatch(favoriteChannel(channel.id));
+        }
+    }, [isFavorite, unfavoriteChannel, favoriteChannel, channel.id]);
+
+    const toggleFavoriteRef = useRef<HTMLButtonElement>(null);
+
+    const removeTooltipLink = useCallback(() => {
+        // Bootstrap adds the attr dynamically, removing it to prevent a11y readout
+        toggleFavoriteRef.current?.removeAttribute('aria-describedby');
+    }, []);
+
+    if (channelIsArchived) {
+        return null
+    }
+
+    const formattedMessage = isFavorite ? {
+        id: 'channelHeader.removeFromFavorites',
+        defaultMessage: 'Remove from Favorites',
+    } : {
+        id: 'channelHeader.addToFavorites',
+        defaultMessage: 'Add to Favorites',
+    };
```

Comment:
Not sure why this is working without the `t` function, but let`s define this outside of the component using `defineMessages`. That way we make sure it is picked by the i18n-extract, and we don't re-define the same object over and over on every render.

Here we can have then something like: `const formattedMessage = isFavorite ? messages['unfavorite'] : messages['favorite'];`

Reply:
I changed it a bit